### DISCUSSION TITLE
Discrepancy in docs?

### DISCUSSION THREAD

**Discussion by dabernathy89** [byu-oit/openapi-enforcer#144] [Q&A] [answered]
Hey, I could be misunderstanding something; please let me know if so. But I'm noticing what appears to be a discrepancy around the return type of `Operation.response()`. [In this section](https://openapi-enforcer.com/api/components/operation.html#request), we see:

> Returns: An [EnforcerResult](https://openapi-enforcer.com/api/enforcer-result.html) that resolves to an object with properties body, header, and schema. If the body passed in was an object then the body result will also be an object, not a JSON string.

However, the [EnforcerResult page](https://openapi-enforcer.com/guide/enforcer-result.html) indicates that the returned object will have the properties `error`, `value`, and `warning`. This is what I'm seeing when I test the code, as well.

Is this a discrepancy or am I missing some context?

**Comment by Gi60s** [comment, marked as answer]
Hello @dabernathy89. I think I understand what you're asking, but if not then please feel free to ask again.

The `Operation.response()` function does return an `EnforcerResult` object. The `EnforcerResult.value` property will contain an object with the `body`, `header`, and `schema` properties.

There are a few ways to get the `value` out of the `EnforcerResult` object. The example given for the [Operation.response()](https://openapi-enforcer.com/api/components/operation.html#response) uses [index desctructuring](https://openapi-enforcer.com/guide/enforcer-result.html#destructure-by-index) to get out the value out of the `EnforcerResult` object.

Let me know if I've answered the question for you or not.

Thanks.

**Reply by dabernathy89** [reply]
Thanks @Gi60s - thanks! That adds some clarity and I did get everything working fine locally. I think perhaps I was tripped up mainly by the specific phrase "resolves to an object with properties body, header, and schema". Conceptually this skips over actual structure of `EnforcerResult`.
